src/kernel/src/fs/mod.rs

```diff
@@ -272,6 +273,27 @@ impl Fs {
         Ok(written.into())
     }
 
+    fn sys_writev(self: &Arc<Self>, i: &SysIn) -> Result<SysOut, SysErr> {
+        let fd: i32 = i.args[0].try_into().unwrap();
+        let ptr: *const libc::iovec = i.args[1].into();
+        let len: usize = i.args[2].try_into().unwrap();
+
+        let td = VThread::current().unwrap();
+        let file = td.proc().files().get(fd).ok_or(SysErr::Raw(EBADF))?;
+        let mut written = 0;
+        for i in 0..len - 1 {
+            let buf: &[u8] = unsafe {
+                std::slice::from_raw_parts(
+                    ptr.offset(i.try_into().unwrap()).read().iov_base as *const u8,
+                    ptr.offset(i.try_into().unwrap()).read().iov_len,
+                )
+            };
+            written = written + file.write(buf as _, Some(&td))?;
```

Comment:
I just looked at this today and I don't think this is correct. Currently, our read and write-related functions don't align with FreeBSD, i.e. file.write
doesn't take an iovec, but just a regular slice, it will have to be refactored.

Reply:
Actually, looking at this now, it's less incorrect that I thought, you call file.write in a loop. But it should still be redone using vectored writes.